packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -107,7 +107,7 @@ function InserterMenu( {
 
 	const onClose = useCallback( () => {
 		// if should replace but didn't insert any block
```

Comment:
// If should replace but didn't insert any block

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/34964/commits/59666727f6ec81241d4eb6977cc63d2feb3dddb2 - turns out I was checking for common words incorrectly